src/main/java/bio/terra/service/tabulardata/google/BigQueryPdao.java

```diff
@@ -106,8 +111,39 @@ public void createDataset(Dataset dataset) {
 
             bigQueryProject.createDataset(datasetName, dataset.getDescription());
             for (DatasetTable table : dataset.getTables()) {
-                bigQueryProject.createTable(datasetName, table.getRawTableName(), buildSchema(table, true));
-                bigQueryProject.createTable(datasetName, table.getSoftDeleteTableName(), buildSoftDeletesSchema());
+                PartitionStrategy partitionStrategy = table.getPartitionStrategy();
+                Column partitionColumn = table.getPartitionColumn();
+                Schema rawSchema = buildSchema(table, true);
+
+                if (partitionStrategy == null) {
+                    bigQueryProject.createUnpartitionedTable(datasetName, table.getRawTableName(), rawSchema);
+                } else {
+                    if (partitionStrategy == PartitionStrategy.INTCOLUMN) {
```

Comment:
If we made a class structure for the partition strategies, we could have them self-generate, instead of using this logic.

Reply:
The tricky part here is that `TimePartitioning` and `RangePartitioning` don't share a superclass themselves, so it's tough to abstract over a method that returns one or the other. We could have an interface like:
```java
interface PartitionMode {
  void setPartition(TableDefinition tableDef)
}
```
And then the internals could no-op or call the appropriate setter. That'd blur the lines pretty heavily between DAO and PDAO lands, though.

Reply:
I guess this would work:
```java
interface PartitionMode {
  TimePartitioning getTimePartitioning()
  RangePartitioning getRangePartitioning()
}
```
And the different subclasses return `null` for the method(s) that they don't know about.